table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1394,9 +1470,82 @@ public void clearStashTokenRangeSnapshot(String stashId) {
         _stashTableDao.clearTokenRanges(stashId);
     }
 
+    @Timed (name = "bv.emodb.table.AstyanaxTableDAO.listMetadataChangedTables", absolute = true)
+    @Override
+    public Iterator<ChangedTable> listMetadataChangedTables(DateTime fromInclusive, DateTime toExclusive, LimitCounter limit) {
+        checkArgument(limit.remaining() > 0, "Limit must be > 0");
+        checkArgument(fromInclusive != null, "from date cannot be null");
+        checkArgument(toExclusive != null, "to date cannot be null");
+
+        LocalDate fromDate = LocalDate.of(fromInclusive.withZone(DateTimeZone.UTC).getYear(), fromInclusive.withZone(DateTimeZone.UTC).getMonthOfYear(),
+                fromInclusive.withZone(DateTimeZone.UTC).getDayOfMonth());
+        LocalDate toDate = LocalDate.of(toExclusive.withZone(DateTimeZone.UTC).getYear(), toExclusive.withZone(DateTimeZone.UTC).getMonthOfYear(),
+                toExclusive.withZone(DateTimeZone.UTC).getDayOfMonth());
+        final int numberOfDays = (int) fromDate.until(toDate, ChronoUnit.DAYS);
+        if (numberOfDays > 30) {
+            throw new IllegalArgumentException("Specified from and to date range is greater than 30 days which is not supported. Specify a smaller range.");
+        }
+
+        List<LocalDate> dates = Stream.iterate(fromDate, d -> d.plusDays(1))
+                .limit(numberOfDays)
+                .collect(Collectors.toList());
+
+        List<ChangedTable> changedTables = Lists.newArrayList();
+        for (LocalDate date : dates) {
+            String dateKey = date.toString();
+            Map<String, Object> recordMap = _backingStore.get(_systemTableMetadataChanges, dateKey, ReadConsistency.STRONG);
+            if (recordMap != null) {
+                ObjectMapper mapper = Jackson.newObjectMapper();
+                Object tableInfo = recordMap.get("tables");
+                if (tableInfo != null) {
+                    Map<String, Object> tableInfoMap = mapper.convertValue(tableInfo, Map.class);
+                    for (Map.Entry<String, Object> tableInfoEntry : tableInfoMap.entrySet()) {
+                        String tableName = tableInfoEntry.getKey();
+                        Object timeUpdateType = tableInfoEntry.getValue();
+                        Map<String, Object> timeUpdateTypeMap = mapper.convertValue(timeUpdateType, Map.class);
+                        if (timeUpdateTypeMap == null) {
+                            throw new IllegalStateException("Returned data is not in the expected format");
+                        }
+                        timeUpdateTypeMap.forEach((key, value) -> changedTables.add(new ChangedTable(tableName, dateKey, key, value.toString())));
+                    }
+                }
+            }
+        }
+
+        return changedTables.iterator();
+    }
+
+    @Override
+    public void writeTableMetadata(String name, TableMetadataAttributes attribute) {
```

Comment:
This method name is very confusing since there is already another method called `updateTableMetadata()`.  Maybe change the names to reflect the this is a metadata event table and not the metadata table.